packages/frontend/src/components/ui/PraiseButton.tsx

```diff
@@ -0,0 +1,42 @@
+import { classNames } from '../../utils/';
+
+type PraiseButtonVariant = '' | 'disabled' | 'outline' | 'round';
+
+interface PraiseButtonProps {
+  id?: string;
+  type?: 'button' | 'submit' | 'reset';
+  variant?: PraiseButtonVariant;
+  classes?: string;
+  children?:
+    | JSX.Element
+    | string
+    | (string | JSX.Element)[]
+    | string[]
+    | JSX.Element[];
+  onClick?: () => void;
+}
+
+export const PraiseButton = ({
+  id,
+  type,
+  variant = '',
+  classes,
+  children,
+  onClick,
+}: PraiseButtonProps): JSX.Element => {
+  const isDisabled = variant === 'disabled';
+  const variantClass =
+    variant === '' || !variant ? 'praise-button' : `praise-button-${variant}`;
```

Comment:
We should avoid creating classnames dynamically. See Tailwind docs for reference:

https://tailwindcss.com/docs/content-configuration

<img width="776" alt="image" src="https://user-images.githubusercontent.com/9698363/183370624-2bfb5ef0-35c5-4c53-97fe-59b993cd3009.png">

Reply:
Also, avoid using the global classnames using `@apply`, especially when they are only used in one component. Instead, define local variables for the variants:

```
const outline = {
  px-4 py-2 …
}
```